File: playpen/swigtest/libsrc/Sphere.cc
// -*- C++ -*-
//
// ======================================================================
//
//                           Brad T. Aagaard
//                        U.S. Geological Survey
//
// {LicenseText}
//
// ======================================================================
//


#include "Sphere.hh"

#include <iostream>
#include <stdexcept>

geometry::Sphere::Sphere(void) :
  _radius(0)
{}
geometry::Sphere::~Sphere(void)
{}

void
geometry::Sphere::radius(const double value) {
  if (value <= 0)
    throw std::runtime_error("Radius must be positive.");
  _radius = value;
}
double
geometry::Sphere::radius(void) const {
  return _radius;
}

void
geometry::Sphere::view(void) const {
  std::cout << "Sphere:\n"
	    << "  radius: " << _radius << "\n"
	    << "  color: " << _color << "\n"
	    << "  id: " << _id << "\n"
	    << std::endl;
}


// End of file

File: playpen/swigtest/python/Sphere.py
#!/usr/bin/env python
#
# ----------------------------------------------------------------------
#
#                           Brad T. Aagaard
#                        U.S. Geological Survey
#
# <LicenseText>
#
# ----------------------------------------------------------------------
#

from Shape import Shape
from geometry import Sphere as ModuleSphere

# Sphere class
class Sphere(Shape, ModuleSphere):

  # INVENTORY //////////////////////////////////////////////////////////

  class Inventory(Shape.Inventory):
    """Python object for managing Sphere facilities and properties."""

    import pyre.inventory

    radius = pyre.inventory.float("radius", default=1.0)
    radius.meta['tip'] = "Radius of sphere."


  # PUBLIC METHODS /////////////////////////////////////////////////////

  def __init__(self, name="sphere"):
    """
    Constructor.
    """
    Shape.__init__(self, name)
    ModuleSphere.__init__(self)
    return


  # PRIVATE METHODS ////////////////////////////////////////////////////

  def _configure(self):
    """
    Setup members using inventory.
    """
    Shape._configure(self)

    self.radius(self.inventory.radius)
    return


# FACTORIES ////////////////////////////////////////////////////////////

def shape():
  return Sphere()


# End of file 
